nullaway/src/main/java/com/uber/nullaway/handlers/RestrictiveAnnotationHandler.java

```diff
@@ -142,4 +143,26 @@ public NullnessHint onDataflowVisitMethodInvocation(
     }
     return NullnessHint.UNKNOWN;
   }
+
+  @Override
+  public NullnessHint onDataflowVisitFieldAccess(
+      FieldAccessNode node,
+      Symbol symbol,
+      Types types,
+      Context context,
+      AccessPath.AccessPathContext apContext,
+      AccessPathNullnessPropagation.SubNodeValues inputs,
+      AccessPathNullnessPropagation.Updates updates) {
+    CodeAnnotationInfo codeAnnotationInfo = getCodeAnnotationInfo(context);
+    // with the generated-as-unannotated option enabled, we want to ignore
+    // annotations in generated code
+    if (config.treatGeneratedAsUnannotated() && codeAnnotationInfo.isGenerated(symbol, config)) {
+      return NullnessHint.UNKNOWN;
+    }
+    if (codeAnnotationInfo.isSymbolUnannotated(symbol, config)
+        && Nullness.hasNullableAnnotation(symbol, config)) {
+      return NullnessHint.HINT_NULLABLE;
+    }
+    return NullnessHint.UNKNOWN;
```

Comment:
Two comments.
* It seems only the `symbol` and `context` parameters are being used here.  Could we get rid of the other ones?
* To me, it seems there is enough common logic within this method and `onOverrideMayBeNullExpr` that it may be worth extracting out to a helper method.  It would take the appropriate parameters and return a `boolean` indicating whether the field should be treated as `@Nullable`, and then this method could convert that to a `NullnessHint`.

Reply:
Also, what is the reason this method returns a `NullnessHint` rather than a `boolean`?

Reply:
- For the parameters, we could get rid of the other ones, but I am trying to make this callback as analogous as possible to `onDataflowVisitMethodInvocation`, since I feel they both might be used in similar ways, and that one provides all that information. We could always trim the number of parameters for now and re-add them as handlers need them if there is a cost, but otherwise it's a bit weird that we have all the dataflow information available when looking at methods, but almost none on field accesses. 
- Let me take a look, it isn't that much shared logic with `onOverrideMayBeNullExpr`, actually, but this method is very much the same logic as `onDataflowVisitMethodInvocation`, so at least those two could be refactored together and maybe the four lines of the conditional could also be shared with `onOverrideMayBeNullExpr`.
- `NullnessHint` is used in `onDataflowVisitMethodInvocation` and it makes sense for this callback to be analogous. The idea behind the type is to give a way for a handler to say "this is generally @Nullable" (e.g. for a method, from a library model) vs "this usage is definitely non-null" (e.g. because of a contract or some property of the dataflow), without the order in which those two determinations apply depending on the order of the handlers. Now that I think about it, that means the usage in `AccessPathNullnessPropagation` is incomplete, since I am not allowing `NullnessHint.FORCE_NONNULL` to override `mayBeNullFieldFromType(...)`

Reply:
> * For the parameters, we could get rid of the other ones, but I am trying to make this callback as analogous as possible to `onDataflowVisitMethodInvocation`, since I feel they both might be used in similar ways, and that one provides all that information. We could always trim the number of parameters for now and re-add them as handlers need them if there is a cost, but otherwise it's a bit weird that we have all the dataflow information available when looking at methods, but almost none on field accesses.

Ok, it's fine to leave the parameters there.

> * Let me take a look, it isn't that much shared logic with `onOverrideMayBeNullExpr`, actually, but this method is very much the same logic as `onDataflowVisitMethodInvocation`, so at least those two could be refactored together and maybe the four lines of the conditional could also be shared with `onOverrideMayBeNullExpr`.

Ok.  The shared logic I saw was around dealing with generated code.  We have so many subtle cases around treatment of unannotated code, generated code, etc. that to me it seems best to share the logic, though it's a small amount of code.

> * `NullnessHint` is used in `onDataflowVisitMethodInvocation` and it makes sense for this callback to be analogous. The idea behind the type is to give a way for a handler to say "this is generally @nullable" (e.g. for a method, from a library model) vs "this usage is definitely non-null" (e.g. because of a contract or some property of the dataflow), without the order in which those two determinations apply depending on the order of the handlers. Now that I think about it, that means the usage in `AccessPathNullnessPropagation` is incomplete, since I am not allowing `NullnessHint.FORCE_NONNULL` to override `mayBeNullFieldFromType(...)`

Ok, sounds good.

Reply:
> The shared logic I saw was around dealing with generated code. We have so many subtle cases around treatment of unannotated code, generated code, etc. that to me it seems best to share the logic, though it's a small amount of code.

You were right, this was _very_ simplifiable in the end! 😁